Common/include/containers/CLookUpTable.hpp

```diff
@@ -0,0 +1,160 @@
+      error_msg.append("' is not in the lookup table.");
+      SU2_MPI::Error(error_msg, CURRENT_FUNCTION);
+    }
+    return index;
+  }
+
+/* get the pointer to the column data of the table (density, temperature, source terms, ...) */
+inline const su2double* GetDataP(const std::string& name_var) {
+    int ix_var = GetIndexOfVar(name_var);
+
+    su2double* tableDataRow(table_data[ix_var]); 
+
+    return tableDataRow;
+  }
+
+  /*!
+   * \brief find the table limits, i.e. the minimum and maximum values of the 2 independent
+   * controlling variables (progress variable and enthalpy). We put the values in the variables
+   * limits_table_prog[2] and limit_table_enth[2]. 
+   * \param[in] name_prog - the string name for the first controlling variable
+   * \param[in] name_enth - the string name of the second controlling variable 
+   */
+  void FindTableLimits(const std::string& name_prog, const std::string& name_enth);
+
+  void IdentifyUniqueEdges();
+
+  void LoadTableRaw(const std::string& file_name_lut);
+
+  void ComputeInterpCoeffs(const std::string& name_prog, const std::string& name_enth);
+
+  void GetInterpMatInv(const su2double* vec_x, const su2double* vec_y, std::array<unsigned long,3>& point_ids,
+                       su2activematrix& interp_mat_inv);
+
+  void GetInterpCoeffs(su2double val_x, su2double val_y, su2activematrix& interp_mat_inv,
+                       std::array<su2double,3>& interp_coeffs);
+
+  su2double Interpolate(const su2double* val_samples, std::array<unsigned long,3>& val_triangle,
+                        std::array<su2double,3>& val_interp_coeffs);
+
+  unsigned long FindNearestNeighborOnHull(su2double val_enth, su2double val_prog, const std::string& name_prog, const std::string& name_enth);
+
+  bool IsInTriangle(su2double val_x, su2double val_y, unsigned long val_id_triangle, const std::string& name_prog,
+                    const std::string& name_enth);
+
+  inline su2double TriArea(su2double x1, su2double y1, su2double x2, su2double y2, su2double x3, su2double y3) {
+    return abs((x1 * (y2 - y3) + x2 * (y3 - y1) + x3 * (y1 - y2)) * 0.5);
+  }
+
+ public:
+  CLookUpTable(const std::string& file_name_lut, const std::string& name_prog, const std::string& name_enth);
+
+  void PrintTableInfo();
+
+  unsigned long LookUp_ProgEnth(const std::string& val_name_var, su2double* val_var, su2double val_prog, su2double val_enth,
+                                const std::string& name_prog, const std::string& name_enth);
+
+  unsigned long LookUp_ProgEnth(std::vector<std::string>& val_names_var, std::vector<su2double*>& val_vars, su2double val_prog,
+                                su2double val_enth, const std::string& name_prog, const std::string& name_enth);
+
+  unsigned long LookUp_ProgEnth(std::vector<std::string>& val_names_var, std::vector<su2double>& val_vars, su2double val_prog,
+                                su2double val_enth, const std::string& name_prog, const std::string& name_enth);
```

Comment:
Document the interface. And what is the difference between the 2nd and 3rd LookUps?
Actually the 3 are slightly different...